### PR TITLE
feat: Refactor LDAR-Sim main script

### PR DESCRIPTION
**Pull request by ThomasGalesloot**
# Pull Request Key Information

## Reason for change

The LDAR-SIm main script became too long and hard to maintain.

## What was changed

This change refactors the main script into smaller method using the new SimulationManager class to maintain state through class variables. 

## Intended Purpose

The main script will now be easier to read and maintain as it is no longer monolithic.

## Level of version change required

N/A

## Testing Completed

Manually ran simple test case 1 on master and after the change and used E2E testing code to compare results, confirmed results were identical: 
[main_vs_new_testing.zip](https://github.com/user-attachments/files/16514686/main_vs_new_testing.zip)

All unit tests pass:
[unit_test_results.txt](https://github.com/user-attachments/files/16514684/unit_test_results.txt)

All End to End tests pass: 
[V4_simple_non_repairable_emissions_dac922076552adcca422bf0c3471343e22073a70.log](https://github.com/user-attachments/files/16514694/V4_simple_non_repairable_emissions_dac922076552adcca422bf0c3471343e22073a70.log)
[V4-simple_repairable_emissions_dac922076552adcca422bf0c3471343e22073a70.log](https://github.com/user-attachments/files/16514695/V4-simple_repairable_emissions_dac922076552adcca422bf0c3471343e22073a70.log)

## Target Issue

N/A

## Additional Information

N/A

### CODE REVIEW

File: LDAR_Sim/src/simulation/simulation_manager.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+import gc
+import multiprocessing as mp
+import os
+import shutil
+import sys
+from datetime import date
+from pathlib import Path
+from typing import Tuple
+
+import pandas as pd
```
Comment (SEAJang) [inline LDAR_Sim/src/simulation/simulation_manager.py:31]: Nit: move pandas import up and separate from ldar-sim specific imports, and alphabetize the above imports

File: LDAR_Sim/src/simulation/simulation_manager.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+            shutil.rmtree(self.out_dir)
+        os.makedirs(self.out_dir)
+        if write_parameters:
+            input_manager.write_parameters(self.out_dir / Output_Files.PARAMETER_FILE)
+
+    def check_generator_files(self) -> bool:
+
+        if os.path.exists(self.generator_dir):
+            print(rm.GEN_WARNING_MSG)
+        print(rm.INIT_INFRA)
```
Comment (SEAJang) [inline LDAR_Sim/src/simulation/simulation_manager.py:154]: Nit: could we move this INIT_INFRA print statement after the pregen/preseed - specifically to be beginning of def setup_virtual_world

File: LDAR_Sim/src/simulation/simulation_manager.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+        if self.preseed_random:
+            self.emis_preseed_val, self.force_remake_gen = gen_seed_emis(
+                self.simulation_count, self.generator_dir
+            )
+
+    def setup_virtual_world(self) -> None:
```
Comment (SEAJang) [inline LDAR_Sim/src/simulation/simulation_manager.py:161]: Nit: Misnomer, as it's only setting up the infrastructure and not the weather/daylight as well. I would rename it to setup_infrastructure.

**Review by SEAJang — APPROVED** [review]
Great job. 
I would say this series of updates would be a patch once all the updates are included..

File: LDAR_Sim/src/simulation/simulation_manager.py (hunk 1 above)
Comment (ThomasGalesloot) [reply LDAR_Sim/src/simulation/simulation_manager.py:31]: I ran the formatter organize imports here this is how they are automatically organized. I propose we use that functionality to organize imports from now on for consistency

File: LDAR_Sim/src/simulation/simulation_manager.py (hunk 2 above)
Comment (ThomasGalesloot) [reply LDAR_Sim/src/simulation/simulation_manager.py:154]: Will do. This is where it was before though - just a note.

File: LDAR_Sim/src/simulation/simulation_manager.py (hunk 3 above)
Comment (ThomasGalesloot) [reply LDAR_Sim/src/simulation/simulation_manager.py:161]: Done

File: LDAR_Sim/src/simulation/simulation_manager.py (hunk 1 above)
Comment (SEAJang) [reply LDAR_Sim/src/simulation/simulation_manager.py:31]: Can you add that work practice to the readme as a separate PR? thanks!

Comment (ThomasGalesloot) [reply LDAR_Sim/src/simulation/simulation_manager.py:31]: Yes

**Pull request merged**